### ISSUE TITLE
Memory leak in ProjectInstance based builds

### ISSUE THREAD

**Issue by cdmihai** [Microsoft/msbuild#2075]
https://github.com/Microsoft/msbuild/pull/1955 introduced a memory leak. Apparently the unique global property strings are leaking into MSBuild's `Microsoft.Build.OpportunisticIntern` string cache.

Code path on an out of proc node:

```
 	Microsoft.Build.OpportunisticIntern.BucketedPrioritizedStringList.InterningToString Line 687	C#
 	Microsoft.Build.OpportunisticIntern.InternableToString Line 175	C#
 	Microsoft.Build.OpportunisticIntern.CharArrayToString Line 199	C#
	Microsoft.Build.InterningBinaryReader.ReadString Line 154	C#
 	Microsoft.Build.BackEnd.NodePacketTranslator.NodePacketReadTranslator.Translate Line 166	C#
 	Microsoft.Build.Execution.ProjectPropertyInstance.FactoryForDeserialization Line 277	C#
 	Microsoft.Build.BackEnd.NodePacketTranslator.NodePacketReadTranslator.Translate<Microsoft.Build.Execution.ProjectPropertyInstance> Line 425	C#
 	Microsoft.Build.BackEnd.NodePacketTranslator.NodePacketReadTranslator.TranslateDictionary<Microsoft.Build.Collections.PropertyDictionary<Microsoft.Build.Execution.ProjectPropertyInstance>, Microsoft.Build.Execution.ProjectPropertyInstance> Line 586	C#
 	Microsoft.Build.Execution.ProjectInstance.TranslateMinimalState Line 1750	C#
 	Microsoft.Build.Execution.ProjectInstance.Microsoft.Build.BackEnd.INodePacketTranslatable.Translate Line 1744	C#
 	Microsoft.Build.Execution.ProjectInstance.ProjectInstance Line 454	C#
 	Microsoft.Build.Execution.ProjectInstance.FactoryForDeserialization Line 1952	C#
 	Microsoft.Build.BackEnd.NodePacketTranslator.NodePacketReadTranslator.Translate<Microsoft.Build.Execution.ProjectInstance> Line 425	C#
 	Microsoft.Build.BackEnd.BuildRequestConfiguration.Translate Line 882	C#
 	Microsoft.Build.BackEnd.BuildRequestConfiguration.BuildRequestConfiguration Line 262	C#
 	Microsoft.Build.BackEnd.BuildRequestConfiguration.FactoryForDeserialization Line 905	C#
 	Microsoft.Build.BackEnd.NodePacketFactory.PacketFactoryRecord.DeserializeAndRoutePacket Line 108	C#
 	Microsoft.Build.BackEnd.NodePacketFactory.DeserializeAndRoutePacket Line 65	C#
 	Microsoft.Build.Execution.OutOfProcNode.Microsoft.Build.BackEnd.INodePacketFactory.DeserializeAndRoutePacket Line 385	C#

```

**Comment by pritish** [comment]
Hi @cdmihai any update?

**Comment by cdmihai** [comment]
Should have been fixed by #2098
Please open if you detected a regression.
